pkg/conversion/capi2mapi/machine_test.go

```diff
@@ -0,0 +1,82 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package capi2mapi
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	capibuilder "github.com/openshift/cluster-api-actuator-pkg/testutils/resourcebuilder/cluster-api/core/v1beta1"
+	capabuilder "github.com/openshift/cluster-api-actuator-pkg/testutils/resourcebuilder/cluster-api/infrastructure/v1beta2"
+	"github.com/openshift/cluster-capi-operator/pkg/conversion/test/matchers"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+var _ = Describe("capi2mapi Machine conversion", func() {
+	var (
+		capiMachineBase = capibuilder.Machine()
+	)
+
+	type capi2MAPIMachineConversionInput struct {
+		machineBuilder   capibuilder.MachineBuilder
+		expectedErrors   []string
+		expectedWarnings []string
+	}
+
+	var _ = DescribeTable("capi2mapi convert CAPI Machine/InfraMachine/InfraCluster to a MAPI Machine",
+		func(in capi2MAPIMachineConversionInput) {
+			_, warns, err := FromMachineAndAWSMachineAndAWSCluster(
+				in.machineBuilder.Build(),
+				capabuilder.AWSMachine().Build(),
+				capabuilder.AWSCluster().Build(),
+			).ToMachine()
+			Expect(err).To(matchers.ConsistOfMatchErrorSubstrings(in.expectedErrors),
+				"should match expected errors while converting CAPI resources to MAPI Machine")
+			Expect(warns).To(matchers.ConsistOfSubstrings(in.expectedWarnings),
+				"should match expected warnings while converting CAPI resources to MAPI Machine")
+		},
+
+		// Base Case.
+		Entry("With a Base configuration", capi2MAPIMachineConversionInput{
+			machineBuilder:   capiMachineBase,
+			expectedErrors:   []string{},
+			expectedWarnings: []string{},
+		}),
+		Entry("With unsupported Version", capi2MAPIMachineConversionInput{
+			machineBuilder:   capiMachineBase.WithVersion(ptr.To("v1.1.1")),
+			expectedErrors:   []string{"spec.version: Invalid value: \"v1.1.1\": version is not supported"},
+			expectedWarnings: []string{},
+		}),
+		Entry("With unsupported NodeDrainTimeout", capi2MAPIMachineConversionInput{
+			machineBuilder:   capiMachineBase.WithNodeDrainTimeout(ptr.To(metav1.Duration{Duration: 1 * time.Second})),
+			expectedErrors:   []string{"spec.nodeDrainTimeout: Invalid value: v1.Duration{Duration:1000000000}: nodeDrainTimeout is not supported"},
+			expectedWarnings: []string{},
+		}),
+		Entry("With unsupported NodeVolumeDetachTimeout", capi2MAPIMachineConversionInput{
+			machineBuilder:   capiMachineBase.WithNodeVolumeDetachTimeout(ptr.To(metav1.Duration{Duration: 1 * time.Second})),
+			expectedErrors:   []string{"spec.nodeVolumeDetachTimeout: Invalid value: v1.Duration{Duration:1000000000}: nodeVolumeDetachTimeout is not supported"},
+			expectedWarnings: []string{},
+		}),
+		Entry("With unsupported NodeDeletionTimeout", capi2MAPIMachineConversionInput{
+			machineBuilder:   capiMachineBase.WithNodeDeletionTimeout(ptr.To(metav1.Duration{Duration: 1 * time.Second})),
```

Comment:
tiniest nit: we could use a const for the timeout

Reply:
Talking to @JoelSpeed regarding test cases, my understanding is that the preferred way of setting this up is with inline values rather than variables, to have a gauge at the input and output without extra lookups/search of values in variables, especially if they are tiny values like this and not big structs & such.
Which I think I kind of agree with.

Reply:
Yeah I think I'd keep this inline so it's easy to gauge (and change) individual test cases when needed

Reply:
Ah ok - that makes sense! :)